src/fondue/Fondue.js

```diff
@@ -649,60 +649,93 @@ export default class Fondue {
 	get featureChars() {
 		const { cmap, GSUB } = this._font.opentype.tables;
 
+		// Human readable names for lookup types
+		const lookupTypes = {
+			1: "Single Substitution",
+			2: "Multiple Substitution",
+			3: "Alternate Substitution",
+			4: "Ligature Substitution",
+			5: "Contextual Substitution",
+			6: "Chained Contexts Substitution",
+			7: "Extension Substitution",
+			8: "Reverse Chaining Contextual Single Substitution",
+		};
+
 		if (!GSUB) return {};
 
 		function letterFor(glyphid) {
 			return cmap.reverse(glyphid).unicode;
 		}
 
-		function parseLookup(lookup, script, lang, feature, currentAllGlyphs) {
-			if (!(feature.featureTag in currentAllGlyphs)) {
-				currentAllGlyphs[feature.featureTag] = {
-					type: lookup.lookupType,
-					input: [],
-					alternateCount: [],
-				};
+		// [1,2,3] + [3,4,5] = [1,2,3,4,5]
+		function mergeUniqueCoverage(existing, addition) {
+			return [...new Set([...(existing || []), ...addition])];
+		}
+
+		// Returns glyphs that are mapped directly to characters for
+		// this coverage. If a glyph maps to another glyph, it's
+		// ignored.
+		// If only a specific rangeRecord needs to be processed, e.g.
+		// for lookup type 3, you can pass the desired index.
+		function charactersFromGlyphs(coverage, index) {
+			let results = [];
+
+			if (!coverage.glyphArray) {
+				let records;
+				if (index >= 0) {
+					records = coverage.rangeRecords.filter(
+						(_, i) => index !== i
+					);
+				} else {
+					records = coverage.rangeRecords;
+				}
+				// Glyphs in start/end ranges
+				for (const range of records) {
+					for (
+						let g = range.startGlyphID;
+						g < range.endGlyphID + 1;
+						g++
+					) {
+						const char = letterFor(g);
+						if (char !== undefined) {
+							results.push(char);
+						}
+					}
+				}
+			} else {
+				// Individual glyphs
+				results = coverage.glyphArray
+					.filter((g) => letterFor(g) !== undefined)
+					.map(letterFor);
 			}
+			return results;
+		}
+
+		function parseLookup(lookup) {
+			const parsedLookup = {
+				type: lookup.lookupType,
+				typeName:
+					lookupTypes[lookup.lookupType] || "Unknown lookup type",
```

Comment:
Shouldn't this function fail if an unknown lookup type is passed? It seems like this function is not doing much if you pass in an unsupported lookup type, so it might as well fail more clearly IMO. WDYT?